### ISSUE TITLE
Copy or move content to another workspace

### ISSUE THREAD

**Issue by EnterMan123** [toeverything/AFFiNE#6743]
### Description

Copy or move documents to another workspace via menu options
For example, from workspace A to workspace B

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Wraithers** [comment]
Yes, this is super valuable after setting up templates or preset docs in a hub workspace, and then moving/duplicating them into another workspace!

Right now I'm having to Export the doc, Import it into the new workspace and then fill / fix all the tables from scratch 🥹

**Comment by dsent** [comment]
This is quite an important feature. One of the strong advantages of Notion is the ability to share/move/copy/duplicate content between different workspaces. People can copy what they liked, as well as reorganize their stuff in different ways, with different workspaces for sharing with different people, working on particular topics etc.

**Comment by fungs** [comment]
This is super important for me, in particular, because a workspace does not have a private incubation section like AppFlowy. I'm starting conceptually in a local or private workspace and want to move over docs to a shared workspace. I'd consider this essential for a paid service.

**Comment by fungs** [comment]
Is there a way to export a single doc? that could be an intermediate step.

**Comment by IveTian** [comment]
> Is there a way to export a single doc? that could be an intermediate step.有没有办法导出单个文档？这可能是一个中间步骤。

Hi there. In the latest version (0.18.1), AFFiNE already supports single doc export to help you. 
Step1. Enter the doc.
Step2. Click the "More" button which is at the doc's title on the right side.
Step3. Select "Export -> Export to Snapshot" in the dropdown menu.
Step4. Find "Import" at AFFiNE's sidebar and click.
Step5. Select "Snapshot" and you can select your snapshot file which you expoted just now to import as a single doc.

I hope this answer helps.

**Comment by IveTian** [comment]
Hi there. The [import improvement] in version 0.18.1 will help you! so make sure you've updated to the latest version.

if you wanna export and import the single AFFiNE doc, you can follow this step:
Step1. Enter the doc.
Step2. Click the "More" button which is at the doc's title on the right side.
Step3. Select "Export -> Export to Snapshot" in the dropdown menu.
Step4. Find "Import" at AFFiNE's sidebar and click.
Step5. Click "Snapshot" button and you can select your snapshot file which you exported just now to import as a single doc.

I hope this answer helps.

**Comment by fungs** [comment]
Thanks @IveTian, I have trouble to find that function.

I'm running v0.18.1 (Linux AppImage) and the web version. If I click the three dots next to the title (I assume this is the "more" button), I can see the export subentry, but it only lists HTML, PNG, Markdown and Print as options. I don't see any snapshot entry.

**Comment by graydawnc** [comment]
> Thanks @IveTian, I have trouble to find that function.
> 
> I'm running v0.18.1 (Linux AppImage) and the web version. If I click the three dots next to the title (I assume this is the "more" button), I can see the export subentry, but it only lists HTML, PNG, Markdown and Print as options. I don't see any snapshot entry.

Sorry, maybe the app image web version hasn’t been updated with the new import and export ui entry yet. 
If you’re using v0.18.0, can you see the snapshot import and export toggle in Settings -> About AFFiNE ? 

<img width="954" alt="Screenshot 2024-11-21 at 22 06 37" src="https://github.com/user-attachments/assets/5b65a45a-39c5-4261-a95c-67a8696c362c">


After enabling the funtion, you can find the snapshot item and sub menu in the More menu, where you can perform the import and export.

<img width="702" alt="Screenshot 2024-11-21 at 22 07 36" src="https://github.com/user-attachments/assets/7981e2a9-80fa-459d-a7d6-0f44fbde34d7">

Or you can also find the import snapshot entry in import dialog after enabling the snapshot import and export funciton.

<img width="611" alt="Screenshot 2024-11-21 at 22 07 59" src="https://github.com/user-attachments/assets/20723e76-db54-4ad6-aba2-a51a3a62f2ba">

In version 0.18.1, the snapshot import/export toggle will be removed, and you will no longer need to enable the toggle to use this feature. 

After removing the toggle, you can find the "Export Snapshot" option in More Menu -> Export 

<img width="608" alt="Screenshot 2024-11-21 at 21 58 28" src="https://github.com/user-attachments/assets/0806f356-ddca-492f-99e3-53fe92a04a22">

Or in the Share Menu

<img width="427" alt="Screenshot 2024-11-21 at 21 58 40" src="https://github.com/user-attachments/assets/47c74156-a1fc-46b3-bded-f1b24eeccd37">

And you can find the snapshot import option in the Import Dialog.

<img width="640" alt="Screenshot 2024-11-21 at 21 58 49" src="https://github.com/user-attachments/assets/bd19c39d-8500-4501-804a-1389cb677aa6">

**Comment by fungs** [comment]
Thanks @donteatfriedrice , indeed I was still running 0.18.0 and had to enable that setting first. Seems to work now!

**Comment by fungs** [comment]
Hi, as there isn't a final statement before closing this issue: is there a copy-between-workspaces function now, or is the export-import workaround the long-term solution?

**Comment by inside-mo** [comment]
Same question here: Hardly seems like a good final or medium-term solution, but is this going to be it for now?

**Comment by cippinocipollino** [comment]
Hi, when u plan to have a serious function of Copy/Move bethween workspaces? Are you working about this feature or is not important for you? We must wait a lot or you are near to release? Thanks

**Comment by russellmilliner** [comment]
What a convoluted method to move content between workspaces.  Ran into this problem on my first day.  Might be a showstopper issue for me going forward with this product.

**Comment by PixPMusic** [comment]
The work above, being done in https://github.com/chrisjensen/AFFiNE/tree/feature/move-docs, does seem to be a much more thought out way to move content between workspaces, ala Notion.

This is pretty core to how I manipulate my notes. Stuff comes in to my main scratch workspace, then gets moved to it's resting place as it gets finalized and promoted for consumption.

**Comment by phxyz12** [comment]
The snapshots approach showed earlier isn't available in the current version, it got removed. It all looks like a lock-in thing which makes it a no-go to migrate 20+ year PKM into.
